tests/test_transformation/test_logcp_transformer.py

```diff
@@ -0,0 +1,155 @@
+    assert transformer.variables is None
+    assert transformer.C == "auto"
+    # test fit attr
+    assert transformer.variables_ == ["Age", "Marks"]
+    assert transformer.n_features_in_ == 5
+    assert all(transformer.C_ == learned_c)
+    # test transform output
+    pd.testing.assert_frame_equal(X, transf_df)
+
+
+@pytest.mark.parametrize(
+    "log_base, exp_age, exp_marks, learned_c", _params_test_automatic_find_variables
+)
+def test_log_base_plus_user_passes_var_list(
+    log_base, exp_age, exp_marks, learned_c, df_vartypes
+):
+    transformer = LogCpTransformer(base=log_base, variables="Age")
+    X = transformer.fit_transform(df_vartypes)
+
+    # expected output
+    transf_df = df_vartypes.copy()
+    transf_df["Age"] = exp_age
+
+    # test init params
+    assert transformer.base == log_base
+    assert transformer.variables == "Age"
+    assert transformer.C == "auto"
+    # test fit attr
+    assert transformer.variables_ == ["Age"]
+    assert transformer.n_features_in_ == 5
+    assert all(transformer.C_ == learned_c[0])
+    # test transform output
+    pd.testing.assert_frame_equal(X, transf_df)
+
+
+@pytest.mark.parametrize("log_base, exp_age, c_age", _params_test_inverse_transform)
+def test_inverse_transform(log_base, exp_age, c_age, df_vartypes):
+
+    transformer = LogCpTransformer(base=log_base, variables="Age")
+    X_t = transformer.fit_transform(df_vartypes)
+    X_dt = transformer.inverse_transform(X_t)
+
+    # expected output
+    transf_df = df_vartypes.copy()
+    transf_df["Age"] = exp_age
+
+    # test init params
+    assert transformer.base == log_base
+    assert transformer.C == "auto"
+    # test fit attr
+    assert transformer.variables_ == ["Age"]
+    assert transformer.n_features_in_ == 5
+    assert all(transformer.C_ == c_age)
+    # test transform output
+    pd.testing.assert_frame_equal(X_dt, transf_df)
+
+
+def test_error_if_base_value_not_allowed():
+    with pytest.raises(ValueError):
+        LogCpTransformer(base="other")
+
+
+def test_fit_raises_error_if_na_in_df(df_na):
+    # test case 3: when dataset contains na, fit method
+    with pytest.raises(ValueError):
+        transformer = LogCpTransformer()
+        transformer.fit(df_na)
+
+
+def test_transform_raises_error_if_na_in_df(df_vartypes, df_na):
+    # test case 4: when dataset contains na, transform method
+    with pytest.raises(ValueError):
+        transformer = LogCpTransformer()
+        transformer.fit(df_vartypes)
+        transformer.transform(df_na[["Name", "City", "Age", "Marks", "dob"]])
+
+
+def test_error_if_df_contains_negative_values(df_vartypes):
+    # test error when data contains negative values
+    df_neg = df_vartypes.copy()
+    df_neg.loc[1, "Age"] = -df_vartypes["Age"].min() - 2
+
+    # when variable contains negative value, transform
+    with pytest.raises(ValueError):
+        transformer = LogCpTransformer()
+        transformer.fit(df_vartypes)
+        transformer.transform(df_neg)
+
```

Comment:
do we need a test for when user specifies a c and the variable + the c is still negative during fit()? We have the error catch already, it is just a matter of adding a test.

I anticipate that many users will just make c=1 without even looking at the variable values, so transform seems a bit late to pick up this error.

Reply:
yes, good catch, when `C` is set manually then it is possible to raise this error. I'll add a test for that